docs/pages/kubernetes-access/controls.mdx

```diff
@@ -303,20 +239,295 @@ roleRef:
   name: teleport-impersonation
 subjects:
 - kind: ServiceAccount
-  # this should be changed to the name of the Kubernetes ServiceAccount being used
   name: teleport-serviceaccount
   namespace: default
 ```
 
-Take a look at the example usage in a [Teleport Helm chart](https://github.com/gravitational/teleport/blob/v\(=teleport.version=\)/examples/chart/teleport-cluster/templates/clusterrole.yaml)
+<Notice type="tip">
+
+The [manual methods](./register-clusters.mdx) and [automatic
+methods](./discovery.mdx) for registering Kubernetes clusters with Teleport
+include steps for setting up the Kubernetes RBAC resources that Teleport needs
+to allow access to clusters.
+
+There is usually no need to define these resources manually.
+
+</Notice>
+
+#### Specifying groups and users based on SAML and OIDC traits
+
+Teleport's roles map OIDC claims or SAML attributes using template variables.
+The template variable `{{external.groups}}` will be substituted to with a user's
+group list in a role definition:
+
+```yaml
+kind: role
+version: v6
+metadata:
+  name: group-member
+spec:
+  allow:
+    # For Alice, will be substituted with the list ["admins", "devs"]
+    kubernetes_groups: ["{{external.groups}}"]
+    # For Alice, will be substituted with ["alice@example.com"]
+    kubernetes_users: ["{{external.email}}"]
+```
+
+### `kubernetes_resources`
+
+The `kubernetes_resources` field enables a Teleport role to configure access to
+specific resources in a Kubernetes cluster:
+
+```yaml
+kind: role
+metadata:
+  name: kube-access
+version: v6
+spec:
```

Comment:
you must also specify the `kubernetes_labels`. The access is only granted if the cluster satisfies the `kubernetes_labels` and the pod satisfies the `kubernetes_resources`